### PR TITLE
System test failure: aest_community_fork_SUITE - timeout #3530

### LINKED ISSUE #3530: System test failure: aest_community_fork_SUITE - timeout
```

%%% aest_channels_SUITE: ...
%%% aest_commands_SUITE: .
%%% aest_community_fork_SUITE: 
%%% aest_community_fork_SUITE ==> fork_chain: FAILED
%%% aest_community_fork_SUITE ==> {timeout,
    [{aest_nodes,wait_for_value,6,
         [{file,
              "/home/circleci/aeternity/_build/system_test+test/extras/system_test/common/helpers/aest_nodes.erl"},
          {line,826}]},
     {aest_nodes,loop_for_values,6,
         [{file,
              "/home/circleci/aeternity/_build/system_test+test/extras/system_test/common/helpers/aest_nodes.erl"},
          {line,788}]},
     {aest_community_fork_SUITE,fork_chain,1,
         [{file,
              "/home/circleci/aeternity/_build/system_test+test/extras/system_test/common/aest_community_fork_SUITE.erl"},
          {line,162}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1754}]},
     {test_server,run_test_case_eval1,6,
         [{file,"test_server.erl"},{line,1263}]},
     {test_server,run_test_case_eval,9,
         [{file,"test_server.erl"},{line,1195}]}]}

### PR DESCRIPTION
**Pull request by velzevur**
Fixes #3530

### CODE REVIEW

**Comment by velzevur** [comment]
I've pushed to `system-tests` branch to see if the error still persists.

File: system_test/common/aest_community_fork_SUITE.erl [hunk 1]
```diff
@@ -29,7 +29,7 @@
 
 %=== MACROS ====================================================================
 
--define(MINING_TIMEOUT,   3000).
+-define(MINING_TIMEOUT,   5000).
```
Comment (velzevur) [inline system_test/common/aest_community_fork_SUITE.erl:32]: Actual fix: assumption is the hardware on CI is unreliably slow.

**Comment by velzevur** [comment]
Error still occurs.

**Review by uwiger — APPROVED** [review]

**Comment by velzevur** [comment]
This error did not show up this time. Restarting the tests.

Assumption is that there is a hidden assumption in the test: at the end we want the network to split into two: `node1` and `node2` shall be on one fork and `node3` and `node4` - on the other. Both networks have a mining nodes (`node1` and `node4`) so they could reach that height. The issue comes with `node3` and its connectivity with `node4` - if they talk to each other, then once `node4` reaches a certain height, blocks would propagate to `node3`.

Fix assumption is that `node3` does not talk directly to `node4` and is receiving blocks by it through either `node1`, `node2` or both. Them being on a separate network would mean that they would discard any block from `node4` mined after the forking point. This would mean that `node1` and `node2` perform an eclipse attack on `node3` and the test fails :)

File: system_test/common/aest_community_fork_SUITE.erl (hunk 1 above)
Comment (velzevur) [reply system_test/common/aest_community_fork_SUITE.erl]: Edit: not the fix.

File: system_test/common/aest_community_fork_SUITE.erl [hunk 2]
```diff
@@ -75,7 +75,7 @@
 
 -define(NODE4,
         #{name            => node4,
-          peers           => [node1, node2, node3],
+          peers           => [node3],
```
Comment (velzevur) [inline system_test/common/aest_community_fork_SUITE.erl:78]: Actual fix: let the forking nodes talk to each other

**Review by gorbak25 — APPROVED** [review]

**Review by cytadela8 — APPROVED** [review]

**Pull request merged**